pallets/rewards/src/lib.rs

```diff
@@ -427,4 +427,17 @@ pub mod pallet {
 			Currencies::<T, I>::get(currency_id).0
 		}
 	}
+
+	impl<T: Config<I>, I: 'static> Pallet<T, I>
+	where
+		RewardAccountOf<T, I>: FullCodec + Default,
+	{
+		pub fn list_currencies(
+			account_id: T::AccountId,
+		) -> sp_std::vec::Vec<(T::DomainId, T::CurrencyId)> {
+			StakeAccounts::<T, I>::iter_prefix(account_id)
+				.map(|(currency_id, _)| currency_id)
+				.collect::<sp_std::vec::Vec<(T::DomainId, T::CurrencyId)>>()
```

Comment:
As a hint, I think this could be simple:
```suggestion
				.collect()
```

Reply:
Done, thank you!